From b63587f870d2f46afce1bd803d69d805d913a8ba Mon Sep 17 00:00:00 2001
From: Zoltan Varga <vargaz@gmail.com>
Date: Fri, 27 Nov 2020 18:54:48 -0500
Subject: [PATCH 3/3] Preserve UnmanagedCallersOnlyAttribute on mono, the
 runtime needs it, and it can be removed when using --used-attrs-only.

---
 .../System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml  | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml b/src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml
index 0732f1f65a8..dbf1729c918 100644
--- a/src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml
+++ b/src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml
@@ -688,6 +688,10 @@
 			<field name="FirstChanceException"/>
 		</type>
 
+		<!-- mono_method_has_unmanaged_callers_only_attribute () -->
+		<!-- aot-compiler.c -->
+		<type fullname="System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute"/>
+
 		<type fullname="Mono.NullByRefReturnException">
 			<!-- marshal-ilgen.c:emit_invoke_call -->
 			<method signature="System.Void .ctor()" />
-- 
2.17.1

